src/api/transaction.go

```diff
@@ -259,3 +260,76 @@ func getRawTx(gateway Gatewayer) http.HandlerFunc {
 		return
 	}
 }
+
+func verifyTxHandler(gateway Gatewayer) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != http.MethodPost {
+			wh.Error405(w)
+			return
+		}
+
+		if r.Header.Get("Content-Type") != "application/json" {
+			wh.Error415(w)
+			return
+		}
+
+		var v struct {
+			EncodedTransaction string `json:"encoded_transaction"`
+		}
+
+		if err := json.NewDecoder(r.Body).Decode(&v); err != nil {
+			wh.Error400(w, err.Error())
+			return
+		}
+
+		tx, err := decodeAndVerifyTx(gateway, v.EncodedTransaction)
+		if err != nil {
+			wh.Error422(w, err.Error())
+			return
+		}
+
+		inputs, err := gateway.GetUxBalances(tx.In)
+		if err != nil {
+			wh.Error503(w, err.Error())
+			return
+		}
+
+		txRsp, err := NewCreatedTransaction(tx, inputs)
+		if err != nil {
+			wh.Error400(w, err.Error())
+			return
+		}
+
+		rsp := struct {
+			Transaction interface{} `json:"transaction"`
+		}{
+			Transaction: txRsp,
+		}
+
+		wh.SendJSONOr500(logger, w, rsp)
+	}
+}
+
+func decodeAndVerifyTx(gateway Gatewayer, encodedTx string) (*coin.Transaction, error) {
+	var tx coin.Transaction
+	b, err := hex.DecodeString(encodedTx)
+	if err != nil {
+		return nil, err
+	}
+
+	tx, err = coin.TransactionDeserialize(b)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, o := range tx.Out {
+		if o.Address.Null() {
+			return nil, errors.New("Transaction.Out contains an output sending to an empty address")
+		}
+	}
+
+	if err := gateway.VerifySingleTxnAllConstraints(&tx); err != nil {
```

Comment:
Use Txn when abbreviating Transaction instead of Tx, use Tx only for dbutil.Tx/bolt.Tx now,

should be method `decodeTxn`, then call gateway.VerifyTxnVerbose, which will call visor.VerifyTxnVerbose which will call VerifySingleTxnAllConstraints and also extract the ux balances for the inputs to make the verbose transaction.